docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+provides a simple solution to track your finances and achieve your financial goals. 
+
+#### Main functions:
+* Adding, deleting and editing transaction entries
+* Viewing your transactions and filtering by certain fields
+* Saving your data into a csv file
+* Setting budgeting goals
+
+#### Purpose and target reader
+
+This guide is designed for potential engineers  and professors who may want to who may want to understand how rainyDay is implemented. 
+We assume that you already have a certain level of technical knowledge as we will be going into detail of the technical aspects
+of rainyDay. <br> If you are a user looking for instructions on how to use rainyDay, please refer to our [UserGuide](https://ay2223s2-cs2113t-t09-1.github.io/tp/UserGuide.html)
+
+#### How to use this guide
+
+{todo, talk about the diagrams, formatting (markdown etc)}
+
+## Design
+
+> 💡**Tip:** The .puml files used to create diagrams in this document can be found [here](https://github.com/AY2223S2-CS2113T-T09-1/tp/tree/master/docs/diagrams). 
+> Refer to the PlantUML Tutorial at se-edu/guides to learn how to create and edit diagrams.
+
+### Architecture
+
+#### Main components of the architecture
+- `Main`: Initializes the components in the correct sequence, and connects them up with each other
+- `Parser`: breaks down inputs passed from user
+- `Command`: executes the appropriate commands
+- `Userdata`: Data input by the user
+- `UI`: the user interface
+- `Storage`: reads data from and writes data to the hard disk file
+
+The diagram given below explains the high-level design of rainyDay, and how the components are related to one another.
+
+![architecture.png](images%2Farchitecture.png)
+
+The Sequence Diagram below shows the timeline of how the different components will interact with one another, we will
+illustrate a 'delete' command, e.g. `delete 1`.
+
+![ArchitectureSequence.png](images%2FArchitectureSequence.png)
+
+### Modules component
+
+### Data component
+
+{insert diagrams}
+
+- stores the financial report data, such as financial statement objects
+
+- Monthly expenditures are stored in the form of a Hash Table, mapping the Year and Month to the expenditures
+
+#### Design considerations
+
+- To prevent slow processing of the program when there are many entries, a Hash Table is used to keep track of the
+  expenditures for the month.
+  - We believe that the "Set Budget" feature will be commonly used as it is 1 of the key aspects of a Financial Tracker. 
+    Hence, it would be feasible to implement an additional storage 
+    such that common addition operations and retrieval of information can be done in Amortised O(1).
+  - The Key used is the number of months from the Year 0000, or more precisely calculated by (Year * 12 + Month). 
+    This is to allow for fast processing and collision-free information collection.
+
+
+### Command component
+
+{insert diagrams}
```

Comment:
line 80:
either remove the line or insert the diagram
its a greater in to leave the line there than to have no diagram (according to TA)